File: IPM/P2/app/lib/models/medication.dart
class Medication {
  final int? id;
  final String name;
  final double dosage;
  final String startDate;
  final int treatmentDuration;
  final int patientId;

  // Constructor
  Medication({
    this.id,
    required this.name,
    required this.dosage,
    required this.startDate,
    required this.treatmentDuration,
    required this.patientId,
  });

  // JSON to object
  factory Medication.fromJson(Map<String, dynamic> json) {
    return Medication(
      id: json['id'],
      name: json['name'],
      dosage: json['dosage'],
      startDate: json['start_date'],
      treatmentDuration: json['treatment_duration'],
      patientId: json['patient_id'],
    );
  }

  // Object to JSON
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'dosage': dosage,
      'start_date': startDate,
      'treatment_duration': treatmentDuration,
      'patient_id': patientId,
    };
  }

  @override
  String toString() {
    return 'Medication(id: $id, name: $name, dosage: $dosage, startDate: $startDate, treatmentDuration: $treatmentDuration, patientId: $patientId)';
  }
}

File: IPM/P1/src/entities/medication.py
from typing import Optional

from gi.repository import GObject


class Medication(GObject.GObject):
    def __init__(self, id: Optional[int], name: str, dosage: float, start_date: str, treatment_duration: int, patient_id: int) -> None:
        super().__init__()
        self.id = id
        self.name = name
        self.dosage = dosage
        self.start_date = start_date
        self.treatment_duration = treatment_duration
        self.patient_id = patient_id